scripts/logging/dictionary/database_gen.py

```diff
@@ -264,6 +267,14 @@ def main():
 
     database = LogDatabase()
 
+    if args.build_header:
+        with open(args.build_header) as f:
+            for l in f:
+                match = re.match(r'\s*#define\s+BUILD_VERSION\s+([^\s]*)', l)
```

Comment:
Are we certain the BUILD_VERSION can't have whitespace in it, even with custom specified BUILD_VERSION?

Perhaps we should match on

BUILD_VERSION\s+(^\s.*)

Reply:
exactly the reason I marked this PR `DNM`, cause I got the same though and wanted to verify this.

Build version is allowed to have whitespaces, so I just pushed an update that allows this.